### PR TITLE
Rdcc 1922 3

### PR DESCRIPTION
**Pull request by KOTLAPRA23**
### JIRA link (if applicable) ###

RDCC-1922

### Change description ###

feat:merging the lrd api changes to master

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

File: config/owasp/suppressions.xml [hunk 1]
```diff
@@ -1,12 +1,32 @@
...
+   file name: oauth2-oidc-sdk-7.1.1.jar
+   ]]></notes>
+    <packageUrl regex="true">^pkg:maven/com\.nimbusds/oauth2\-oidc\-sdk@.*$</packageUrl>
+    <cve>CVE-2007-1651</cve>
+    <cve>CVE-2007-1652</cve>
+  </suppress>
+  <suppress>
+    <notes><![CDATA[
+   file name: tomcat-embed-core-9.0.38.jar
+   ]]></notes>
```
Comment (abhijitdiwan) [inline config/owasp/suppressions.xml:21]: please try find alternative as it is new cve.

File: src/integrationTest/java/uk/gov/hmcts/reform/lrdapi/util/LrdApiClient.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+import static uk.gov.hmcts.reform.lrdapi.util.JwtTokenUtil.generateToken;
+
+@Slf4j
+@PropertySource(value = "/integrationTest/resources/application.yml")
+public class LrdApiClient {
+
+    private static final String APP_BASE_PATH = "/refdata/location/orgServices";
+
+    private static final String JWT_TOKEN = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI"
+        + "6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c";
```
Comment (abhijitdiwan) [inline src/integrationTest/java/uk/gov/hmcts/reform/lrdapi/util/LrdApiClient.java:33]: sonar will complain.please avoid hard code

File: src/main/java/uk/gov/hmcts/reform/lrdapi/controllers/LrdApiController.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+                    message = "Internal Server Error"
+            )
+    })
+    @GetMapping(produces = APPLICATION_JSON_VALUE)
+    public ResponseEntity<Object> retrieveOrgServiceDetailsByServiceCodeOrCcdCaseType(
+        @RequestParam(value = "serviceCode",required = false) String serviceCode,
+        @RequestParam(value = "ccdCaseType",required = false) String ccdCaseType) {
+        log.info("inside retrieveServiceCodeDetailsByServiceCodeOrCcdCaseType");
+        List<LrdOrgInfoServiceResponse> lrdOrgInfoServiceResponse = null;
+        if (!StringUtils.isBlank(serviceCode) && !StringUtils.isBlank(ccdCaseType)) {
```
Comment (abhijitdiwan) [inline src/main/java/uk/gov/hmcts/reform/lrdapi/controllers/LrdApiController.java:72]: minor comment here i think isNotBlank is there StringUtils.isNotBlank

File: config/owasp/suppressions.xml (hunk 1 above)
Comment (KOTLAPRA23) [reply config/owasp/suppressions.xml:21]: we have got approval for that

File: src/main/java/uk/gov/hmcts/reform/lrdapi/controllers/LrdApiController.java (hunk 3 above)
Comment (KOTLAPRA23) [reply src/main/java/uk/gov/hmcts/reform/lrdapi/controllers/LrdApiController.java:72]: ok . will change

File: src/integrationTest/java/uk/gov/hmcts/reform/lrdapi/util/LrdApiClient.java (hunk 2 above)
Comment (KOTLAPRA23) [reply src/integrationTest/java/uk/gov/hmcts/reform/lrdapi/util/LrdApiClient.java:33]: for integration test used hardcode token to mock the s2s call like same as PRD. Sonar will not complain

Comment (abhijitdiwan) [reply src/integrationTest/java/uk/gov/hmcts/reform/lrdapi/util/LrdApiClient.java:33]: public static String generateDummyS2SToken(String serviceName) {
        return Jwts.builder()
            .setSubject(serviceName)
            .setIssuedAt(new Date())
            .signWith(SignatureAlgorithm.HS256, TextCodec.BASE64.encode("AA"))
            .compact();
    }

Comment (KOTLAPRA23) [reply src/integrationTest/java/uk/gov/hmcts/reform/lrdapi/util/LrdApiClient.java]: done

**Review by abhijitdiwan — APPROVED** [review]

**Review by AdilOoze — APPROVED** [review]

**Pull request merged**